openmdao/core/problem.py

```diff
@@ -1053,10 +1053,16 @@ def pre_run_check(self):
         raise a readable error for the user."""
 
         # New message if you forget to run setup first.
-        if not self.root.fd_options.locked:
+        if not self.root._order_set:
```

Comment:
I think we need a separate flag that indicates if there have been any configuration changes without setup being called.  _order_set is set to True whenever set_order is called.  set_order can be called directly by the user so _order_set does not ensure that setup was run.  Also, we're only checking the top level group, but we need to make sure that no config changes have occurred anywhere in the tree.  I think there should be a flag like _config_changed or something like that that we can set in _probdata if any system has config changes that would require us to re-run setup. Then we can check that flag here.  We'll need to add properties like we talked about earlier in order to set that flag.